### ISSUE TITLE
Serverless-python-layers: No file matches include / exclude patterns 

### ISSUE THREAD

**Issue by Abhijith-Nagaraja** [serverless/serverless#8878]
**Describe the issue**
No matter what changes I make to `serverless.yaml`, the following error doesn't solve when I add layers to python lambda

```No file matches include / exclude patterns ```

**How to reproduce?**

My folder structure is

```
serverless-test
|_lambdas
|____handler.py
|_layers
|____common
|_________somefunction.py
```

```
service: serverless-test

frameworkVersion: '2'

provider:
  name: aws
  runtime: python3.8
  lambdaHashingVersion: 20201221

  stage: test
  region: us-west-2

functions:
  hello:
    handler: lambdas/handler.hello
```

This works fine. Now as soon as I add a layer, I get the following error

`No file matches include / exclude patterns`

```
service: serverless-test

frameworkVersion: '2'

provider:
  name: aws
  runtime: python3.8
  lambdaHashingVersion: 20201221

  stage: test
  region: us-west-2

functions:
  hello:
    handler: lambdas/handler.hello
    layers:
      - {Ref: CommonLambdaLayer}

layers:
  common:
    path: layers/common
    name: common-module
    description: common set of functions
```

I also tried adding include and exclude patterns. But it didn't solve my problem

```
service: serverless-test

frameworkVersion: '2'

provider:
  name: aws
  runtime: python3.8
  lambdaHashingVersion: 20201221

  stage: test
  region: us-west-2

package:
  individually: true
  exclude: 
    - ./**
  include:
    - ./lambdas/**

functions:
  hello:
    handler: lambdas/handler.hello
    layers:
      - {Ref: CommonLambdaLayer}

layers:
  common:
    path: layers/common
    name: common-module
    description: common set of functions
    package:
      include:
        - ./**
```

I also tried being very specific

```
service: serverless-test

frameworkVersion: '2'

provider:
  name: aws
  runtime: python3.8
  lambdaHashingVersion: 20201221

  stage: test
  region: us-west-2

package:
  individually: true
  exclude: 
    - ./**

functions:
  hello:
    handler: lambdas/handler.hello
    layers:
      - {Ref: CommonLambdaLayer}
    package:
      exclude:
        - ./**
      include:
        - ./lambdas/handler.py

layers:
  common:
    path: layers/common
    name: common-module
    description: common set of functions
    package:
      exclude:
        - ./**
      include:
        - ./layers/common/somefunction.py
```
**Provide (in below placeholder) FULL name and output of the command that exposes the problem.**
`sls deploy --verbose`

```
Serverless: Excluding development dependencies...
Serverless: Excluding development dependencies...
 
  Serverless Error ---------------------------------------
 
  No file matches include / exclude patterns
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          linux
     Node Version:              12.18.2
     Framework Version:         2.21.1
     Plugin Version:            4.4.2
     SDK Version:               2.3.2
     Components Version:        3.6.0
```
**Provide (in below placeholder) output of serverless --version**
2.21.1

**Comment by pgrzesik** [comment]
Hello @Abhijith-Nagaraja, thanks for reporting your issue. I've tried to reproduce your problems, but I believe I might be missing something in your setup. I've prepared a simple repository with reproducible example, but I'm not running into issues and layer is packaged and deployed correctly. Could you take a look here: https://github.com/pgrzesik/simple-lambda-layers-python-reproduce and check out what's different in your setup? 

Thanks in advance :bow:

**Comment by Abhijith-Nagaraja** [comment]
@pgrzesik : my setup is exactly same but one thing I noticed is , when I do `npm install serverless` I get following error in one of its dependency node-gyp. I wonder whether that's causing any issue. The error being

```
make: Entering directory `/var/jenkins/workspace/Serverless-test/serverless-script/node_modules/snappy/build'
  CXX(target) Release/obj.target/snappy/deps/snappy/snappy-1.1.7/snappy-sinksource.o
make: g++: Command not found
make: *** [Release/obj.target/snappy/deps/snappy/snappy-1.1.7/snappy-sinksource.o] Error 127
make: Leaving directory `/var/jenkins/workspace/Serverless-test/serverless-script/node_modules/snappy/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/opt/abhijith/.nvm/versions/node/v12.18.2/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12)
gyp ERR! System Linux 4.9.77-31.58.amzn1.x86_64
gyp ERR! command "/opt/abhijith/.nvm/versions/node/v12.18.2/bin/node" "/opt/abhijith/.nvm/versions/node/v12.18.2/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /var/jenkins/workspace/Serverless-test/serverless-script/node_modules/snappy
gyp ERR! node -v v12.18.2
gyp ERR! node-gyp -v v5.1.0
gyp ERR! not ok 
```

Since serverless installed no matter what, I did not heed to this error. Does this have any impact at all?

**Comment by pgrzesik** [comment]
Hello @Abhijith-Nagaraja - unfortunately I'm not sure if that might be the cause of your issue - do you have the ability to use a standalone installer as described here: https://www.serverless.com/framework/docs/getting-started#install-as-a-standalone-binary ? It's a self-contained binary without dependencies on your local OS/Node.

**Comment by talaikis** [comment]
Just got same problems 
```

> snappy@6.3.5 install /usr/local/lib/node_modules/serverless/node_modules/snappy
rebuild-install \|\| node-gyp rebuild
Command did not exit successfully npm install --no-progress -g npm serverless cross-env exit status 1
```

with previously working deploy script

**Comment by pgrzesik** [comment]
Hello @talaikis - could you please share a bit more about your setup? How are you using `serverless`? What version are you using?

**Comment by Abhijith-Nagaraja** [comment]
@pgrzesik: I was unable to solve this issue. Since my project was a POC or Proof of concept, I changed my runtime from python to Node and it worked seamlessly in spite of having the node-gyp dependency error. I am closing this issue as there is one more open issue which connects to this. 
https://github.com/serverless/serverless/issues/5892

I have added my issue also there, hopefully it will be of some help to someone
